### ISSUE TITLE
Anyone encounter Chrome first time run issue? Thanks.

### ISSUE THREAD

**Issue by Smao0875** [openid/AppAuth-Android#172]
Hi Guys,
When I was playing around with the sample app, I found that there is something unusual when I made request after clearing data of Chrome.

Chrome version: 56.0.2924.87

The activity stack will be after I made auth request:
Task1: MainActivity -> AuthorizationManagementActivity 
Task2: FirstRunActivity (chrome)

Then we hit "ACCEPT & CONTINUE", activity stack won't change.

We choose an google account in the list and click "CONTINUE", activity stack won't change.

We hit "OK, GOT IT", activity stack changes to :
Task1: MainActivity -> AuthorizationManagementActivity 
Task3: Preferences (chrome)
Then a list of items are shown up (I attached an image for you to review)
![screenshot_20170220-110037](https://cloud.githubusercontent.com/assets/12376493/23132755/a2d840bc-f75c-11e6-8365-d21dafd34850.png)

In this screen we couldn't do anything with the existing Google accounts shown in the list.
 -- Nothing happens if we click on these two existing Google accounts.
If we want to add account by choosing "Add accounts", then we will get "account already exist" error.

It only happens at the first time Chrome is being used, should I worry about it or just let it go?

Has anyone else met this kind of situation? I can reproduce this issue on different kinds of devices.

Any suggestions would be greatly appreciated.
Thanks.

**Comment by iainmcgin** [comment]
Sounds like an issue with Chrome rather than AppAuth, due to it only happening after data wipe / first run. I'll notify the Chrome engineers I work with to see if it is a known problem.

**Comment by Smao0875** [comment]
@iainmcgin 
Thanks. Got it.

**Comment by shafty023** [comment]
If anyone has a workaround please share. This is blocking our ability to switch to AppAuth and the deadline for using in-app webviews to authenticate via OAUTH2 is next month.

**Comment by iainmcgin** [comment]
Why is this blocking you? This only happens when the user has never set up Chrome on the device, which is a very unusual circumstance.

**Comment by shafty023** [comment]
@iainmcgin I can reproduce it with the following scenario:
1) Sign into chrome with test2
2) Sign into AppAuth app with test1

We get redirected to some special sign in activity which breaks the context for AppAuth app. This signin activity actually tries adding the google account to the device which is not what we want. We just want to auth with google and have it sign into the google accounts within chrome and not within Android. Regardless, if you sign in at this point you will not be redirected to the original AppAuth app. 

This is going to be very common where people are signed into Chrome with some gmail account but they want to sign into another gmail account with an AppAuth app. So this special signin activity that Chrome is displaying breaks the context to redirect back to the AppAuth app.

**Comment by lquinn** [comment]
@iainmcgin, I don't think it's that unusual for a user with a new device to set up a Gmail account before they set up Chrome.

**Comment by iainmcgin** [comment]
There's unfortunately nothing that can be done to fix the issue from the AppAuth side. I'll bug the Chrome engineers.

**Comment by shafty023** [comment]
@iainmcgin The issue only happens when a user is already signed into chrome. Is there not a way to launch Chrome into incognito mode via AppAuth which would technically get around this problem for now? I know that's not a good solution long term because Google wants the user to only have to sign in once and launching into Incognito mode would break that but it would technically work.

**Comment by iainmcgin** [comment]
No, there is no way to launch an incognito custom tab (or standalone Chrome browser) at present - I confirmed this with the Chrome engineers I work with.

**Comment by shafty023** [comment]
@iainmcgin Did you open a bug with the Chrome team that we can follow by chance? Or is it private/internal?

**Comment by iainmcgin** [comment]
There's no public bug yet, I'm discussing the issue directly with the responsible parties on the Chrome team.

**Comment by shafty023** [comment]
@iainmcgin  Any update on this?

**Comment by iainmcgin** [comment]
A fix has been agreed upon with the Chrome team, I'm not sure exactly when this will land. I'll chase the engineers responsible.

**Comment by ve7jtb** [comment]
One way I found around this issue was to send the user hint to Google from AppAuth.

That seems to get arround the account adding behavior.

I ran into this in a scenario with a G+ enterprise SAML account being used on a device that didn't have it added as a account because it required MDM (Not possible on ChomeOS or desirable in some Android cases).   If the user is redirected to Google and prompted for there email it tries to add the account.   However with the user hint Google immediately redirected the user to Authenticate via SAML and completed the flow logging the user into the browser but not the device.  

I didn't test it with non federated G+ accounts but it might work.

Regards
John B.

**Comment by shafty023** [comment]
@ve7jtb I already send a hint and it doesn't help unfortunately.

**Comment by shafty023** [comment]
@iainmcgin Just wanted to follow up and see if any progress was made.
